src/config/field-overrides/predictor/DataElementCategoryOptionCombo.js

```diff
@@ -0,0 +1,143 @@
+import React, { Component, PropTypes } from 'react';
+import DropDown from '../../../forms/form-fields/drop-down.js';
+import { red500, grey400 } from 'material-ui/styles/colors';
+import { LinearProgress } from 'material-ui';
+
+const DEFAULT = 'default';
+
+const DropDownLoader = ({ msg }) => (
+    <p
+        style={{
+            color: grey400,
+            fontSize: 16,
+            fontWeight: 300,
+            padding: '41px 0 6px',
+            margin: 0,
+        }}
+    >
+        {msg}
+        <LinearProgress style={{ marginTop: 4, height: 2 }} />
+    </p>
+);
+
+class DataElementCategoryOptionCombo extends Component {
+    state = {
+        options: [],
+        loading: false,
+        loadErrorText: '',
+    };
+    getTranslation = this.context.d2.i18n.getTranslation.bind(
+        this.context.d2.i18n
+    );
+
+    // The form builder is mutating objects which causes this.props.model.output
+    // and prevProps.model.output to be equal in the componentDidUpdate hook,
+    // after the output property was updated.
+    // By keeping track of this string literal the changes are detected correctly.
+    prevOutputId = null;
+
+    onChange = event => {
+        const fakeEvent = { target: { value: { id: event.target.value } } };
```

Comment:
This seems confusing. I'd expect `value` to contain the contents of an input, not an object with an `id`. I expect you're just conforming to what the onChange callback expects, but it seems strange to me.

Reply:
You are right on both accounts:
-  I am indeed just conforming to what the onChange callback expects
- Yes it is strange... Let's not implement a form api in this way ever again

Reply:
> Yes it is strange... Let's not implement a form api in this way ever again

Sounds like a good plan 😋